lib/Sema/CSDiagnostics.cpp

```diff
@@ -4416,6 +4416,36 @@ bool ThrowingFunctionConversionFailure::diagnoseAsError() {
   return true;
 }
 
+bool UnnecessaryCoercionFailure::diagnoseAsError() {
+  auto expr = cast<CoerceExpr>(getAnchor());
+  auto sourceRange = SourceRange(expr->getLoc(),
+                                 expr->getCastTypeLoc().getSourceRange().End);
+  
+  if (isa<TypeAliasType>(getFromType().getPointer()) &&
+      isa<TypeAliasType>(getToType().getPointer())) {
+    auto fromTypeAlias = cast<TypeAliasType>(getFromType().getPointer());
+    auto toTypeAlias = cast<TypeAliasType>(getToType().getPointer());
+    // If the typealias are the same, we don't need a warning
```

Comment:
I think this comment needs to go under the `else` branch as here the type aliases _aren't_ same.